### ISSUE TITLE
Repositories for ChildEntities can't be found with string as entityClass (Cordova/Sqljs)

### ISSUE THREAD

**Issue by marcorinck** [typeorm/typeorm#2419]
**Issue type:**

[X] bug report

**Database system/driver:**

[x] `cordova`
[x] `sqljs`

**TypeORM version:**

[x] `latest`

**Steps to reproduce or a small repository showing the problem:**

When trying to get a repository for a child entity in a ionic environment like documented in the doc/example project does not work. I get this error:

````
main.js:3111 Error: No repository for "user" was found. Looks like this entity is not registered in current "default" connection?
    at new RepositoryNotFoundError (vendor.js:183687)
    at SqljsEntityManager.EntityManager.getRepository (vendor.js:69298)
    at Connection.getRepository (vendor.js:110102)
    at getRepository (vendor.js:20178)
```` 
(the stack was created using sqljs while in browser mode, but its the same error message while using cordova) 

For ionic its specifically mentioned in the doc to ask for the repository not with the entity class but its string identifier (because build optimizer) like this :

````typescript
getRepository("user") as Repository<User>;
```` 

When I use the entity class to ask for the repository it works as expected (while in development mode): 

````typescript
getRepository(User);
```` 

The entities look like this: 

````typescript
@Entity("person")
@TableInheritance({column: {type: "varchar", name: "type"}})
export class Person {

  @PrimaryGeneratedColumn()
  id: number;

  @Column({nullable: true})
  firstname: string;
}

@ChildEntity("user")
export class User extends Person {

  @Column()
  token: string;

}
````

**Comment by pleerock** [comment]
can you try to disable mangling in ugly options and try again?

**Comment by pleerock** [comment]
Is this issue only occurs when using `TableInheritance` ? Do you have problems with regular entities?

**Comment by marcorinck** [comment]
oh I thought I made that clear, I'm sorry. This ONLY happens for child entities. Normal entities are fine.

**Comment by pleerock** [comment]
so your issue that you can't find entity by a `string name` right? If yes then why don't you try to use class instead of string?

**Comment by marcorinck** [comment]
Yes, that is right. I did this because its specifically mentioned in the Ionic example project to do so because of the build optimizer used in Ionic in production mode.

**Comment by pleerock** [comment]
try to use class and disable mangle and see how it affects your app

**Comment by marcorinck** [comment]
I just opened an issue in the ionic example project (see above) as I'm unable to do ANY production builds because of a typescript bug which can only be fixed when using TS >=2.7.x which is incompatible with every supported angular version of the current ionic version. 

To reiterate for this issue:
* I don't have any problem when using getRepository(EntityClass) or getRepositry(ChildEntityClass)
* I don't have any problem when using getRepository("nameOfEntity") when nameOfEntitiy refers to a "normal" entity which was defined with Entity decorator
* a repositories can't be found even in dev mode (without any optimizations) when using getRepository("nameOfChildEntity"). The method signature of getRepository suggests that this should work as its defined as getRepository(entityClass: ObjectType<Entity> | string) but it doesn't work when string is the name of a ChildEntity when using Single-table-inheritance.
* as production builds for ionic are currently not possible, I don't know if any of this is true in production mode

**Comment by imnotjames** [comment]
This looks like a duplicate of #2164 - closing as such.  Ionic's mangling in these versions don't work very well with TypeORM.
